src/Microsoft.AspNetCore.Authentication.Facebook/FacebookOptions.cs

```diff
@@ -57,6 +60,6 @@ public string AppSecret
         /// The list of fields to retrieve from the UserInformationEndpoint.
         /// https://developers.facebook.com/docs/graph-api/reference/user
         /// </summary>
-        public IList<string> Fields { get; }
+        public ICollection<string> Fields { get; } = new HashSet<string>();
```

Comment:
The only difference with IList was that it guaranteed ordering, though I don't know if that will matter to any of the providers.

As a guideline an API should return the most general thing it can so that in the future you can return something more specific without breaking people. In this case I'd recommend ICollection over ISet as you don't actually need any of the ISet APIs, you just want it for descriptiveness sake. Doc comments should be sufficient explanation for now.

You can do Scope too if you want, but start a separate PR since it doesn't overlap with any of this work.